### PR TITLE
DTD-4559 Cucumber to scalaTest - Manually write scalaTests for IFS API repo (2)

### PR DESCRIPTION
**Pull request by AlfieMolloy99**
Manually write new scalaTest tests for 31 scenarios. 
LeapYear.feature
MultipeDebtItems EdgeCases.feature
MultipeDebtItems.feature
SuppressionEdgeCases.feature
Supression.feature
Added two fields to DebtCaclation to cover edge case scenarios with default None values.

### CODE REVIEW

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsEdgeCasesFeatureSpec.scala [hunk 1]
```diff
@@ -16,485 +16,995 @@
...
+        amountOnIntDueDuty = 500000,
+        totalAmountIntDuty = 500000,
+        unpaidAmountDuty = 500000,
+        interestOnlyIndicator = false,
+        calculationWindows = Nil
+      )
+      theDebtSummaryWillContain(context, 2, expected2ndDebtSummary)
 
+      And("the 2nd debt summary will have no calculation windows")
+      theDebtSummaryWillNotHaveAnyCalculationWindows(context, 2)
```
Comment (balaganesan) [inline src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsEdgeCasesFeatureSpec.scala:161]: this method doesn't have any implementation

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsEdgeCasesFeatureSpec.scala [hunk 2]
```diff
@@ -16,485 +16,995 @@
...
-      // noPostCodesHaveBeenProvidedForTheCustomer(context)
 
-      When("the debt item is sent to the ifs service")
-      // TODO: Helper 'theDebtItemIsSentToTheIfsService' expects context 'InterestForecastingContext' but this spec uses 'FCStatementOfLiabilityContext'.
-      // Validate whether this scenario should use a different context or whether the helper should be aligned to this spec context.
-      // theDebtItemIsSentToTheIfsService(context)
-
-      Then(
-        "the ifs service will respond with Could not parse body due to requirement failed: originalAmount can be zero or greater, negative values are not accepted"
+    Scenario("9. 2 SA debts where one has an original amount less than zero") { context =>
```
Comment (balaganesan) [inline src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsEdgeCasesFeatureSpec.scala]: include the tag @DTD-2216

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsFeatureSpec.scala [hunk 3]
```diff
@@ -16,387 +16,511 @@
...
+        amountOnIntDueDuty = 400000,
+        totalAmountIntDuty = 400000,
+        unpaidAmountDuty = 400000,
+        interestOnlyIndicator = false,
+        calculationWindows = Nil
+      )
+      theDebtSummaryWillContain(context, 1, expectedDebtSummary)
+
+      And("the debt summary will have no calculation windows")
+      theDebtSummaryWillNotHaveAnyCalculationWindows(context, 1)
```
Comment (balaganesan) [inline src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsFeatureSpec.scala:103]: this method doesn't have any implementation

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsFeatureSpec.scala [hunk 4]
```diff
@@ -16,387 +16,511 @@
...
-      And("no post codes have been provided for the customer")
-      // TODO: Helper 'noPostCodesHaveBeenProvidedForTheCustomer' expects context 'InterestForecastingContext' but this spec uses 'FCStatementOfLiabilityContext'.
-      // Validate whether this scenario should use a different context or whether the helper should be aligned to this spec context.
-      // noPostCodesHaveBeenProvidedForTheCustomer(context)
-
-      When("the debt items is sent to the ifs service")
-      // TODO: Helper 'theDebtItemIsSentToTheIfsService' expects context 'InterestForecastingContext' but this spec uses 'FCStatementOfLiabilityContext'.
-      // Validate whether this scenario should use a different context or whether the helper should be aligned to this spec context.
-      // theDebtItemIsSentToTheIfsService(context)
+    Scenario("Interest Bearing. 1 Payment of 1 debt.") { context =>
```
Comment (balaganesan) [inline src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsFeatureSpec.scala]: update the correct scenario name  "Interest Bearing. 2 Payments of 1 debt."

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SuppressionEdgeCasesFeatureSpec.scala [hunk 5]
```diff
@@ -19,516 +19,1228 @@ package uk.gov.hmrc.test.api.scalatest.specs.ifs
...
+    val context = InterestForecastingContext()
     try test(context)
     finally ()
   }
 
   Feature("Suppression - Edge cases") {
+    val suppressionContext = SuppressionRulesContext()
 
-    ignore("Suppression, interest rate change during suppression") { context =>
+    Scenario(
```
Comment (balaganesan) [inline src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SuppressionEdgeCasesFeatureSpec.scala:50]: add these tags at least in the first scenario @suppression, DTD_2790

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SuppressionEdgeCasesFeatureSpec.scala [hunk 6]
```diff
@@ -19,516 +19,1228 @@ package uk.gov.hmrc.test.api.scalatest.specs.ifs
...
+        amountOnIntDueDuty = 500000,
+        totalAmountIntDuty = 500000,
+        unpaidAmountDuty = 500000,
+        interestOnlyIndicator = false,
+        calculationWindows = Nil
+      )
+      theDebtSummaryWillContain(context, 1, expectedDebtSummary)
+
+      And("the debt summary will have no calculation windows")
+      theDebtSummaryWillNotHaveAnyCalculationWindows(context, 1)
```
Comment (balaganesan) [inline src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SuppressionEdgeCasesFeatureSpec.scala:1244]: include the implementation for this method

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SupressionFeatureSpec.scala [hunk 7]
```diff
@@ -19,729 +19,1361 @@ package uk.gov.hmrc.test.api.scalatest.specs.ifs
...
     try test(context)
     finally ()
   }
 
   Feature("Suppression") {
+    val suppressionContext = SuppressionRulesContext()
 
-    ignore("Suppression - full address postCode") { context =>
+    Scenario(
+      "Suppression, 2 payments on different dates during suppression"
```
Comment (balaganesan) [inline src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SupressionFeatureSpec.scala]: since there are two tags @suppression @DTD-2790 in the feature, so include those in the first scenario.

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsEdgeCasesFeatureSpec.scala (hunk 1 above)
Comment (AlfieMolloy99) [reply src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsEdgeCasesFeatureSpec.scala:161]: addressed under [43e1657]

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsFeatureSpec.scala (hunk 3 above)
Comment (AlfieMolloy99) [reply src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsFeatureSpec.scala:103]: added under [43e1657]

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsEdgeCasesFeatureSpec.scala (hunk 2 above)
Comment (AlfieMolloy99) [reply src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsEdgeCasesFeatureSpec.scala]: tags added under [2dc0462]

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsFeatureSpec.scala (hunk 4 above)
Comment (AlfieMolloy99) [reply src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/MultipeDebtItemsFeatureSpec.scala]: fixed under [2dc0462]

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SuppressionEdgeCasesFeatureSpec.scala (hunk 5 above)
Comment (AlfieMolloy99) [reply src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SuppressionEdgeCasesFeatureSpec.scala:50]: added under [2dc0462]

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SupressionFeatureSpec.scala (hunk 7 above)
Comment (AlfieMolloy99) [reply src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SupressionFeatureSpec.scala]: added with [2dc0462]

File: src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SuppressionEdgeCasesFeatureSpec.scala (hunk 6 above)
Comment (AlfieMolloy99) [reply src/test/scala/uk/gov/hmrc/test/api/scalatest/specs/ifs/SuppressionEdgeCasesFeatureSpec.scala:1244]: method updated under [43e1657]

**Review by balaganesan — APPROVED** [review]
